### ISSUE TITLE
Bug when editing the same full row twice.

### ISSUE THREAD

**Issue by Youngmj1994** [ag-grid/ag-grid#5447]
```
[X] bug report => see 'Providing a Reproducible Scenario'
[] feature request => do not use Github for feature requests, see 'Customers of AG Grid'
[] support request => see 'Requesting Community Support'
```

**Providing a Reproducible Scenario**
1. Go to https://www.ag-grid.com/angular-data-grid/cell-editing-full-row/
2. Double click on a row to Edit. 
3. Click outside of the row to cancel edit
4. Click the same row again to edit and you will be unable to edit. 

**Current behavior**
When editing a full row, there is an error where you edit the row, then click out or cancel the editing, then you try to edit that row again and you cannot edit the row anymore until you refresh the page entirely. 

When looking into the code this is where I discovered the issue
```
RowCtrl.prototype.startRowEditing = function (key, charPress, sourceRenderedCell, event) {
        if (key === void 0) { key = null; }
        if (charPress === void 0) { charPress = null; }
        if (sourceRenderedCell === void 0) { sourceRenderedCell = null; }
        if (event === void 0) { event = null; }
        // don't do it if already editing
        if (this.editingRow) {
            return; //PROBLEM HERE: When attempting to edit a row for a second time this.editingRow is permanently set to true
        }
        this.getAllCellCtrls().forEach(function (cellCtrl) {
            var cellStartedEdit = cellCtrl === sourceRenderedCell;
            if (cellStartedEdit) {
                cellCtrl.startEditing(key, charPress, cellStartedEdit, event);
            }
            else {
                cellCtrl.startEditing(null, null, cellStartedEdit, event);
            }
        });
        this.setEditingRow(true);
    };
```

**Expected behavior**
You should be able to edit a row multiple times. 

**Please tell us about your environment:**
Discovered on Windows, using VSCode and Angular

* **AG Grid version:** X.X.X
Version 28.1.0

* **Browser:**
Using Chrome

* **Language:** TypeScript

**Comment by StephenCooper** [comment]
Thank you for reporting this - much appreciated! 

Based on your description we've now reproduced this issue as a bug in v28.1. This means if you wish to use v28 we would recommend pinning to 28.0.0 explicitly for now until we can get this fixed.

We have added this bug to our backlog and we are tracking it with the following reference and description:
**AG-7118 [Full Row Edit] Cannot Edit the same row twice**

We try to fix bugs from one release to the next, so this should be fixed in the next release or the one after if it was raised too close to the next release date.

You can see the next release date on the product pipeline page:
https://www.ag-grid.com/ag-grid-pipeline/

Now that this issue is recorded in our backlog, we will soon close the GitHub issue because we will not update it when this bug is fixed in a future version of AG Grid. The best way to get an update on this item is to sign up for AG Grid new release notifications. Once a new AG Grid version is released, look for the item reference above (AG-7118) on the changelog page to see if it's resolved in that version:
https://www.ag-grid.com/ag-grid-changelog/

Thanks again for bringing this up with us - we appreciate your time!

**Comment by agrt56** [comment]
I can confirm, that 28.0.0 is working as intended

**Comment by StephenCooper** [comment]
This has now been resolved in the patch release 28.1.1
